eclipse/che-plugin-registry
# feat(generation): Remove the need of bash script, using library generation for all the stuff

### What does this PR do?
Also move out from docker build the generation of files to allow better caching across builds/better dogfooding
Unify everything through build.sh (dogfooding or image build)
Always use digest instead of tags (and verify referenced imaged exists)

Introduce github action caching for yarn and vsix to speed up the execution of actions

related PR for che documentation: https://github.com/eclipse/che-docs/pull/1816

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
Fix https://github.com/eclipse/che/issues/18857


### How to test this PR?
All plug-ins should work as before
Digest of images are used instead of tags

note: devfile to develop che plugin registry is not working on che.openshift.io (too much resources are needed)

here is devfile to try content coming from this PR

http://che.openshift.io/f/?url=https://gist.githubusercontent.com/benoitf/c0454e1acf2a08e6a624c318944f528e/raw/be43a42aa18943eb4b2d09120ca6b2c07afaf5e0/devfile.yaml

```yaml
apiVersion: 1.0.0
metadata:
  generateName: test-new-plugin-registry
projects:
  - name: che-plugin-registry
    source:
      type: github
      location: https://github.com/eclipse/che-plugin-registry
components:
  - id: eclipse/che-theia/next
    type: cheEditor
    registryUrl: 'https://pr-check-815-alpine-che-plugin-registry.surge.sh/v3'
  - id: redhat/java/latest
    type: chePlugin
    registryUrl: 'https://pr-check-815-alpine-che-plugin-registry.surge.sh/v3'    
  - id: vscode/typescript-language-features/latest
    type: chePlugin
    registryUrl: 'https://pr-check-815-alpine-che-plugin-registry.surge.sh/v3'        
```


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

Change-Id: I89e880cc9944f9c570af17ff82dc58f861285931
Signed-off-by: Florent Benoit <fbenoit@redhat.com>
